src/App.js

```diff
@@ -1,23 +1,66 @@
-import logo from './logo.svg';
+import { Row } from "antd";
+import React, {useState} from 'react'
 import './App.css';
+import foodsJSON from "./foods.json";
+import FoodBox from "./components/FoodBox";
+import AddFoodForm from "./components/AddFoodForm";
+import Search from "./components/Search";
 
 function App() {
+
+
+  const [ allFoods, updatedList ] = useState(foodsJSON)
+const [allFilterFoods, setAllFilterFoods] = useState(foodsJSON)
+
+   const addNewFood = (food) => {
+   const newFood = [food, ...allFoods]
+   updatedList(newFood)
+  }
+
+  const searchFood = (str) => {
+    let filteredFood
+    if (str === "") {
+      filteredFood = allFilterFoods
+    } else {
+      filteredFood = allFilterFoods.filter(food => {
+        return food.name.toLowerCase().includes(str.toLowerCase())
+      })
+    }
+    updatedList(filteredFood)
+  }
+
+  const deleteFood = (name) => {
+    const foodInfo = allFoods.filter(food => {
+      return food.name !== name
+    })
+    updatedList(foodInfo)
+  }
+
+ 
+
   return (
     <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+      <div>
+
+        <AddFoodForm createFood={addNewFood}/>
```

Comment:
Here, you need to send the `addNewFood` as props to your component where you are getting the function to create the food and pass the correct values. So, you need to do it like this:
`<AddFoodForm addNewFood={addNewFood}/>`